objects/src/notes/file.rs

```diff
@@ -0,0 +1,50 @@
+use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_processor::DeserializationError;
+
+use super::{Note, NoteDetails, NoteInclusionProof};
+
+// NOTE FILE
+// ================================================================================================
+
+/// A serialized representation of a note.
+pub enum NoteFile {
+    /// The note has not yet been recorded on chain.
+    Details(NoteDetails),
+    /// The note has been recorded on chain.
+    Recorded(Note, NoteInclusionProof),
+}
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for NoteFile {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        match self {
+            NoteFile::Details(details) => {
+                target.write_u8(0);
+                details.write_into(target);
+            },
+            NoteFile::Recorded(note, proof) => {
+                target.write_u8(1);
+                note.write_into(target);
+                proof.write_into(target);
+            },
+        }
```

Comment:
It might be good to start serialization with "magic" 4 byte value (e.g., could be ASCII value of the word `note`). This way, when deserializing an object, we could immediately tell if we are deserializing a note or something else (e.g., an account).

Reply:
Done. I like the idea.